src/utils/specific_label_utils.py

```diff
@@ -119,6 +95,129 @@ def is_insertion_or_replication(word, correct_words_list):
             return True
     return False    
 
+#  --- Number labeling Typos and OCRs ---
+
+def differentiate_errors_in_number_column(data_column: pd.Series, generic_labeled_cell_indices: pd.Index, generic_labeled_dataset: pd.DataFrame, label_func: callable = None, min_value: float = None, max_value: float = None) -> pd.Series:
+    """
+    This function applies a specific labeling function to differentiate errors in number columns.
+    Parameters:
+    - data_column: pd.Series - The column of data to be labeled.
+    - generic_labeled_cell_indices: pd.Index - The indices of the cells that have been labeled in the generic dataset.
+    - generic_labeled_dataset: pd.DataFrame - The dataset containing the generic labels.
+    - function: callable - The function to apply for labeling. It takes as input a single word and returns the ErrorType. The label_number function (which contains a set of rules for distinguishing typos and OCRs for numbers) will never be applied in case a label_func is specified. You can call it from the label_func, if you want.
+    - min_value: float - The minimum value of the number.
+    - max_value: float - The maximum value of the number.
+    """
+    label_column = pd.Series(0, index=data_column.index, dtype=int)
+    flawed_words_series = generic_labeled_dataset.loc[generic_labeled_cell_indices]
+    unique_flawed_words = flawed_words_series.unique()
+
+    typo_word_map = {}
+
+    for word in unique_flawed_words:
+        if label_func:
+            typo_word_map[word] = label_func(word, min_value, max_value) 
+        else:
+            typo_word_map[word] = label_number_with_ocr_or_typo(word, min_value, max_value)
+
+    # Remap results back to the original indices
+    for index, word in flawed_words_series.items():
+        if typo_word_map.get(word, False):
+            label_column.loc[index] = typo_word_map[word]
+
+    return label_column
+
+def label_number_with_ocr_or_typo(word: str | int | float, min_value: float = None, max_value: float = None):
+    """
+    Method, which can be used to label numbers with errors, in case there is no column-specific labeling function for numbers defined.
+    It distinguishes between typos and OCRs for numbers.
+    """
+    
+    word = str(word)
+    if word == 77:
+        print("77 from label_number_with_error")
+    if word[-1] in [".", ",", "-"]: # a number like 8743. that ends with a dot is likely an OCR
+        return ErrorType.OCR.value
+
+    if word[0] == "0" and word[1] != ".": # a number like 08.1 that starts with a 0 is likely an OCR
+        return get_label_for_number_with_0_prefix(word, min_value, max_value)
+
+    if " " in word:
+        return ErrorType.OCR.value
+
+    if is_likely_ocr(word):
```

Comment:
I would either rename this method or move the other OCR checks into it, because the name suggests it is the one method that can differentiate ocr and the logic should be encapsualted here!

Reply:
Okay, I renamed it to `has_letter_mapping_to_ocr_number`